src/main/webapp/app/shared/route/OncokbRoute.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect } from 'react';
+import { RouteProps, Route, RouteComponentProps } from 'react-router';
+
+type CanonicalLinkPropsConfig =
+  | {
+      __canonicalTypeName: 'location-without-url-params';
+    }
+  | {
+      __canonicalTypeName: 'function';
+      getCanonicalEndpoint: (
+        routeProps: RouteComponentProps,
+        removedPropsUrl: string
+      ) => string;
+    }
+  | {
+      __canonicalTypeName: 'location';
+    }
+  | { __canonicalTypeName?: undefined };
+type CanonicalLinkProps = { routeProps: RouteComponentProps } & {
+  canonicalProps: CanonicalLinkPropsConfig | undefined;
+};
+
+function CanonicalLink({ routeProps, canonicalProps }: CanonicalLinkProps) {
+  useEffect(() => {
+    if (!canonicalProps) return;
+    const removedPropsUrl = routeProps.match.path.replace(/\/:.*/, '');
+    const canonicalEndpoint =
+      canonicalProps.__canonicalTypeName === 'location'
+        ? routeProps.location.pathname
+        : canonicalProps.__canonicalTypeName === 'function'
+        ? canonicalProps.getCanonicalEndpoint(routeProps, removedPropsUrl)
+        : canonicalProps.__canonicalTypeName === 'location-without-url-params'
+        ? removedPropsUrl
+        : undefined;
+
+    const linkNode = document.createElement('link');
+
+    linkNode.setAttribute('id', 'canonical');
+    linkNode.setAttribute('rel', 'canonical');
+    linkNode.setAttribute('href', `${location.host}${canonicalEndpoint}`);
+
+    document.head.appendChild(linkNode);
+
+    return () => {
+      document.head.removeChild(linkNode);
```

Comment:
Do you think this is needed?

Reply:
because I didn't add a canonical link for every route so I don't want to confuse the google crawler by having an incorrect link

Reply:
Oh right, nvm. I thought the page is reloaded on every redirect, but we are SPA. In that case, do we need to look for the canonical tag before appending? Or is the dom is start enough to avoid dup?

Reply:
Avoid duplication and to remove the link tag all together if we didn't specify needing one.

Reply:
Sorry how is it doing what you said above?

Reply:
It always deletes the link on a route change, but only adds it back when you specify in the OncokbRoute component that you want a canonical link.